admin/commands/management/add.go

```diff
@@ -97,3 +97,27 @@ func processGlobalAddFlagsWithSocket(cmd connectionGetter, opts AddCommonFlags)
 
 	return serviceName, socket, host, uint16(portI), nil
 }
+
+// Determine if parameters were passed.
+var (
+	usernameParameterSpecified = false
+	passwordParameterSpecified = false
+)
+
+// Types for username and password
+type (
+	username string
+	password string
+)
+
+// AfterApply Change usernameParameterSpecified flag if argument was passed
+func (u username) AfterApply() error {
+	usernameParameterSpecified = true
+	return nil
+}
+
+// AfterApply Change passwordParameterSpecified flag if argument was passed
+func (p password) AfterApply() error {
+	passwordParameterSpecified = true
+	return nil
+}
```

Comment:
seems like these new types can be easily replaced by just pointers https://github.com/alecthomas/kong#pointers

Reply:
This is going to be a quite long comment.

When decided to use pointers https://github.com/alecthomas/kong#pointers, pmm-admin compiled but failed to launch:
`panic: AddProxySQLCommand.Username: unsupported field type *string, perhaps missing a cmd:"" tag?`
since --username/--password are **flags** adding cmd:"" will make parameter as a **subcommand**.
![2022-10-24_19-14](https://user-images.githubusercontent.com/12951615/197598752-9a716e31-4d52-401a-b5e1-cbcc263fb364.png)

I did some research and found simillar problem with pointers in subcommands: https://github.com/alecthomas/kong/issues/196. Issue was solved in latest version which pmm-admin uses, so I jumped into PRs source https://github.com/alecthomas/kong/commit/43616c6a6d5989dd9b3512e80f00ed3337b8b842#diff-679d4747ec9379321e0f2e758c26439f99115a71b4618af9713a65f65d160fb7R1286
The problem is that the fix is only for subcommands, not for flags.

I've replaced

> Cmd *cmd \`cmd:""\`

with
> Cmd   *string \`help:"PostgreSQL username"\`

and the result was the same, as for pmm-admin.
![2022-10-24_20-23](https://user-images.githubusercontent.com/12951615/197600131-8457f9f8-e467-4db2-9d13-adcc8c5d5e64.png)

It looks like Kong's **inner commands** have problems with **pointers for parameters**.

In next step i've tried to add `cmd:""` just to see how the app will behave.
Unfortunately, it doesn't work as expected. Passed empty parameter `--username` and skipped `--password` wasn't nill but equal "" in runtime. 
So pointers won't solve this issue :( 

![2022-10-24_19-24](https://user-images.githubusercontent.com/12951615/197600763-e3b5ffed-5217-419e-a225-b35239630562.png)

Should we stick to the current solution?

Reply:
Yes let's stick on current solution with some TODO that ot should be fixed once kong will start supporting it. And can we create an issues in kongs repository, please.

Reply:
Just checked that main branch with kong source code and it works fine. 
then find out that this code https://github.com/alecthomas/kong/pull/296 hasn't been released yet (merged 20th Sept).

There is no need to create new issue. Will add TODO that it should be fixed after kong version bumping.
Thanks!

Reply:
Great, thank you

Reply:
@pkadej pointers don't work because they have not been release yet and are not part of v0.6.1
With the master branch they work as expected.

I'd opt for using the pointers. We can either wait until a new release is available or we can pin the module to a specific commit and upgrade as soon as the new version is out.
Either way, I tried pinging the maintainers to see when a new release is due https://github.com/alecthomas/kong/pull/296#issuecomment-1290048303

Reply:
v0.7.0 has just been released. We shall upgrade and use pointers.

Reply:
Done. Updated, retested and pushed. Works. Thanks :)